modules/fedimint-wallet-tests/tests/tests.rs

```diff
@@ -230,3 +240,163 @@ async fn peg_outs_support_rbf() -> anyhow::Result<()> {
+    )?));
+
+    match wallet
+        .process_input(
+            &mut dbtx.with_module_prefix(module_instance_id),
+            &input,
+            &fedimint_wallet_server::WalletVerificationCache,
+        )
+        .await
+    {
+        Ok(_) => bail!("Expected peg-in to fail"),
+        Err(e) => {
+            assert!(e.to_string().contains("Unknown block hash in peg-in proof"));
+        }
+    }
+
+    // Mine the rest of the blocks
+    bitcoin
+        .mine_blocks((wallet_config.consensus.finality_delay - 1).into())
+        .await;
+    let block_count = dyn_bitcoin_rpc.get_block_count().await? - 1;
+    sync_wallet_to_block(
+        &mut dbtx.with_module_prefix(module_instance_id),
+        &mut wallet,
+        block_count.try_into()?,
+    )
+    .await?;
+
+    assert_matches!(
+        wallet
+            .process_input(
+                &mut dbtx.with_module_prefix(module_instance_id),
+                &input,
+                &fedimint_wallet_server::WalletVerificationCache,
+            )
+            .await,
+        Ok(_)
+    );
+    dbtx.commit_tx().await;
+    Ok(())
+}
+
+async fn sync_wallet_to_block(
+    dbtx: &mut ModuleDatabaseTransaction<'_>,
+    wallet: &mut fedimint_wallet_server::Wallet,
+    block_count: u32,
+) -> anyhow::Result<()> {
+    for peer in 0..(MINTS / 2 + 1) {
+        let consensus_item = fedimint_wallet_common::WalletConsensusItem::BlockCount(block_count);
+        let peer_id = PeerId::from(peer as u16);
+        wallet
+            .process_consensus_item(dbtx, consensus_item, peer_id)
+            .await?;
+    }
+    Ok(())
+}
+
+const MINTS: usize = 5;
+
+fn build_wallet_server_configs(
+    bitcoin_rpc: BitcoinRpcConfig,
+) -> anyhow::Result<(
+    Vec<fedimint_core::config::ServerModuleConfig>,
+    fedimint_core::config::ClientModuleConfig,
+)> {
+    let peers = (0..MINTS as u16).map(PeerId::from).collect::<Vec<_>>();
+    let wallet_cfg = fedimint_core::module::ServerModuleInit::trusted_dealer_gen(
+        &WalletGen,
+        &peers,
+        &fedimint_core::config::ConfigGenModuleParams::from_typed(WalletGenParams {
+            local: fedimint_wallet_common::config::WalletGenParamsLocal {
+                bitcoin_rpc: bitcoin_rpc.clone(),
+            },
+            consensus: fedimint_wallet_common::config::WalletGenParamsConsensus {
+                network: bitcoin::Network::Regtest,
+                finality_delay: 10,
+                client_default_bitcoin_rpc: bitcoin_rpc.clone(),
+            },
+        })?,
+    );
+    let client_cfg = fedimint_core::config::ClientModuleConfig::from_typed(
+        0,
+        <WalletGen as fedimint_core::module::ServerModuleInit>::kind(),
+        fedimint_core::module::ModuleConsensusVersion(0),
+        fedimint_core::module::ServerModuleInit::get_client_config(
+            &WalletGen,
+            &wallet_cfg[&PeerId::from(0)].consensus,
+        )?,
+    )?;
+    Ok((wallet_cfg.into_values().collect(), client_cfg))
+}
```

Comment:
We can probably deduplicate parts of this function, something similar seems to be needed for each module:

https://github.com/fedimint/fedimint/blob/af41ed842d174ffdd80f5ab274594fcae36fc962/modules/fedimint-mint-server/src/lib.rs#L783-L806